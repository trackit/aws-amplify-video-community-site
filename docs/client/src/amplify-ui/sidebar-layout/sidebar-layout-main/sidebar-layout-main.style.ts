import {MQFablet} from "../../styles/media";
import {css} from "emotion";

export const sidebarLayoutMainStyle = css`
  overflow-x: hidden;
  transition: 0.25s linear box-shadow;
  padding: 1.5rem 0;
  width: 100%;

  &.in-view {
    box-shadow: inset -0.5rem 0 0.5rem -0.5rem rgba(0, 0, 0, 0.5);
  }

  ${MQFablet} {
    &.in-view {
      box-shadow: initial;
    }
  }

  > div {
    min-width: 100vw;
    padding: 0 1rem;

    ${MQFablet} {
      min-width: initial;
      padding: 0 2rem 0 4rem;
    }
  }
`;
