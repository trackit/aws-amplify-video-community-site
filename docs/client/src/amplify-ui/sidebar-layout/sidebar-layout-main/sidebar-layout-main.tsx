import {Component, Host, Prop, h} from "@stencil/core";
import {sidebarLayoutContext} from "../sidebar-layout.context";
import {sidebarLayoutMainStyle} from "./sidebar-layout-main.style";

@Component({tag: "amplify-sidebar-layout-main", shadow: false})
export class AmplifySidebarLayoutMain {
  /*** whether or not the sidebar is in view, provided by `sidebar-layout` */
  @Prop() readonly inView?: boolean;

  render() {
    return (
      <Host
        class={{
          [sidebarLayoutMainStyle]: true,
          "in-view": this.inView === undefined || this.inView,
        }}
      >
        <div>
          <slot />
        </div>
      </Host>
    );
  }
}

sidebarLayoutContext.injectProps(AmplifySidebarLayoutMain, ["inView"]);
