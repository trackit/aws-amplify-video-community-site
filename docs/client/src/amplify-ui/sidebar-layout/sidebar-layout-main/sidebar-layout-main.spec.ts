import {AmplifySidebarLayoutMain} from "./sidebar-layout-main";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-sidebar-layout-main", () => {
  describe("Component logic", () => {
    let amplifySidebarLayoutMain: AmplifySidebarLayoutMain;
    beforeEach(
      () => (amplifySidebarLayoutMain = new AmplifySidebarLayoutMain()),
    );

    it("should init `inView` as `undefined`", () =>
      expect(amplifySidebarLayoutMain.inView).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifySidebarLayoutMain],
            html: `<amplify-sidebar-layout-main />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
