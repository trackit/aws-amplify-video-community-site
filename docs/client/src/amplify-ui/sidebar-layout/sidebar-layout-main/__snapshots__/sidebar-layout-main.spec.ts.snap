// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sidebar-layout-main Render logic should render 1`] = `
<amplify-sidebar-layout-main class="css-1pjk5d in-view">
  <!---->
  <div></div>
</amplify-sidebar-layout-main>
`;
